nifi-nar-bundles/nifi-hubspot-bundle/nifi-hubspot-processors/src/main/java/org/apache/nifi/processors/hubspot/GetHubSpot.java

```diff
@@ -0,0 +1,320 @@
+    public static final String API_BASE_URI = "api.hubapi.com";
+    public static final String HTTPS = "https";
+
+    private volatile WebClientServiceProvider webClientServiceProvider;
+
+    private static List<PropertyDescriptor> createPropertyDescriptors() {
+        final List<PropertyDescriptor> propertyDescriptors = new ArrayList<>(Arrays.asList(
+                ACCESS_TOKEN,
+                CRM_ENDPOINT,
+                LIMIT,
+                WEB_CLIENT_PROVIDER
+        ));
+        return Collections.unmodifiableList(propertyDescriptors);
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        webClientServiceProvider = context.getProperty(WEB_CLIENT_PROVIDER).asControllerService(WebClientServiceProvider.class);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final String accessToken = context.getProperty(ACCESS_TOKEN).getValue();
+        final String endpoint = context.getProperty(CRM_ENDPOINT).getValue();
+
+        final StateMap state = getStateMap(context);
+        final URI uri = createUri(context, state);
+
+        final HttpResponseEntity response = getHttpResponseEntity(accessToken, uri);
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final JsonFactory jsonFactory = objectMapper.getFactory();
+        FlowFile flowFile = session.create();
+
+        flowFile = session.write(flowFile, out -> {
+
+
+            try (JsonParser jsonParser = jsonFactory.createParser(response.body());
+                 final JsonGenerator jsonGenerator = jsonFactory.createGenerator(out, JsonEncoding.UTF8)) {
+                while (jsonParser.nextToken() != null) {
+                    if (jsonParser.getCurrentToken() == JsonToken.FIELD_NAME && jsonParser.getCurrentName().equals("results")) {
+                        jsonParser.nextToken();
+                        jsonGenerator.copyCurrentStructure(jsonParser);
+                    }
+                    String fieldname = jsonParser.getCurrentName();
+                    if ("after".equals(fieldname)) {
+                        jsonParser.nextToken();
+                        Map<String, String> newStateMap = new HashMap<>(state.toMap());
+                        newStateMap.put(endpoint, jsonParser.getText());
+                        updateState(context, newStateMap);
+                        break;
+                    }
+                }
+            }
+        });
+        session.transfer(flowFile, REL_SUCCESS);
```

Comment:
It is possible for the query to return zero results? If so, that condition should be checked before transferring.